src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -148,10 +179,54 @@ Status MediapipeGraphDefinition::create(std::shared_ptr<MediapipeGraphExecutor>&
         return status;
     }
     SPDLOG_DEBUG("Creating Mediapipe graph executor: {}", getName());
-    pipeline = std::make_shared<MediapipeGraphExecutor>(getName(), std::to_string(getVersion()), this->config);
+
+    bool passKfsRequestFlag = false;
+    // Check if we are passing whole KFS request and response
+    status = this->setKFSPassthrough(passKfsRequestFlag);
+    if (!status.ok()) {
+        SPDLOG_DEBUG("Failed to execute mediapipe graph: {} KFS passthrough mode is misconfigured.", getName());
+        return status;
+    }
```

Comment:
Since this is based on config json this could fail at validation stage not during inference.

Reply:
Then we can keep bool member of MediapipeGraphDefinition to store that information and use it during ::create()